pkg/boot/uefi/uefi.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uefi
+
+import (
+	"bytes"
+	"debug/pe"
+	"encoding/binary"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/u-root/u-root/pkg/acpi"
+	"github.com/u-root/u-root/pkg/boot"
+	"github.com/u-root/u-root/pkg/boot/kexec"
+)
+
+const fvEntryImageOffset int64 = 0xA0
+
+var bootExecute = boot.Execute
+var kexecLoad = kexec.Load
+var kexecParseMemoryMap = kexec.ParseMemoryMap
+var getRSDP = acpi.GetRSDP
```

Comment:
In lieu of mocking out, can you implement a qemu like https://github.com/u-root/u-root/blob/master/integration/generic-tests/io_test.go#L17

Reply:
I added a TODO for adding integration test. For unittest, we should still mock functions from external packages as unittest only tests this package.